casper/src/main/resources/PoS.rhox

```diff
@@ -72,35 +117,165 @@ in {
+                  for (@deployerPubKeyBytes <- deployerPubKeyBytesCh) {
+                    if (deployerPubKeyBytes ==  pubkey.hexToBytes()){
+                      new unfAuthKeyCh in {
+                        @RevVault!("unforgeableAuthKey", *unf, *unfAuthKeyCh) |
+                        for (@unfAuthKey <- unfAuthKeyCh){
+                          @vault!("transfer", targetAddress, amount, unfAuthKey, *transferRet)
+                        }
+                      }
+                    } else {
+                      transferRet!((false, "You have not permission to transfer."))
+                    }
+                  }
+                }
+              } |
+              ret!((*unf, unfAddr, vault ,*unfVaultHandle))
+            }
+          }
+        }
+      }|
+      createUnfVault!(*posVaultCh, "$$posPubKey$$") |
+      for (@(posVaultUnf, posVaultAddr, posVault, humanControllHandle) <- posVaultCh){
+      @RevVault!("unforgeableAuthKey", posVaultUnf, *posAuthKeyCh) |
+      for (@posDeployerRevAddress <- posDeployerRevAddressCh;
+           @posDeployerAuthKey    <- posDeployerAuthKeyCh;
+           @posAuthKey <- posAuthKeyCh)  {
+        for (@(true, (coopMultiVault, coopMultiVaultAddr, coopVaultHd)) <<- coopVaultCh) {
           // Creates PoS Rev vault with sum of initial bonds.
           new bondSumCh in {
             @ListOps!("fold", $$initialBonds$$.toList(), 0, *sumFromPair, *bondSumCh) |
             for (@bondSum <- bondSumCh) {
-              @RevVault!("createWithBalance", posRevAddress, bondSum, *posVaultCh)
+              @RevVault!("createWithBalance", posDeployerRevAddress, bondSum, *posDeployerVaultCh)|
+              for (@(true, posDeployerVault)  <- posDeployerVaultCh) {
+                // move all the fund from deployer addr to POS unforgeable vault
+                @posDeployerVault!("transfer", posVaultAddr, bondSum, posDeployerAuthKey, *moveInitialBondCh)
+              }
             }
           } |
           // Selects the initial active validators.
-          pickActiveValidators!($$initialBonds$$, {}, *initialActiveCh) |
-          for (@(true, posVault)  <- posVaultCh;
-               @initialActive     <- initialActiveCh) {
-            new stateCh, pendingRewardsInitializedCh in {
+          pickActiveValidators!($$initialBonds$$.toList(), *initialActiveCh) |
+          for (@initialActive     <- initialActiveCh; 
+               @(true, _) <- moveInitialBondCh) {
+            new stateCh, pendingRewardsInitializedCh, rewardsInfo, calTotalActiveBonds, calTotalBonds, calTotalWithdraw, calTotalCommittedRewards in {
               // Initializes PoS state structure.
               stateCh!({
                 // Map[PublicKey, Int] - each validator stake
                 "allBonds"        : $$initialBonds$$,
-                // List[PublicKey]     - the active validators
+                // Set[PublicKey]     - the active validators
```

Comment:
I changed the dataType to `Set` because I want to delete the active validator immediately when slashing happen.

Reply:
It is easier to just use the `delete` method in `Set`